aws/credential_cache.go

```diff
@@ -3,17 +3,74 @@ package aws
 import (
 	"context"
 	"sync/atomic"
+	"time"
 
+	sdkrand "github.com/aws/aws-sdk-go-v2/internal/rand"
 	"github.com/aws/aws-sdk-go-v2/internal/sync/singleflight"
 )
 
+// CredentialsCacheOptions are the options
+type CredentialsCacheOptions struct {
+	// Provider is the CredentialProvider implementation to be wrapped by the CredentialCache.
+	Provider CredentialsProvider
```

Comment:
Should `Provider` be an option since its a required parameter in the constructor? Having it as an option seems like a way to get unexpected behavior having it in both places. 

I think it would be better to be a unexported member on the `CredentialsCache` type instead a part of the options.